### PR TITLE
SPSH-594: Seeding-Skripte Versionieren und Ausführung vermerken

### PR DESCRIPTION
**Pull request by DPDS93CT**
# Description
Changes behavior of seeding:
- evaluates whether certain seeding-files were seeded already based on their SHA265-checksum
- seeding files now are located in sub-directories for better structure when working incremental (e.g. seeding/dev/01)
- already successfully seeded files will be skipped
- already failed files will be skipped
- improved logging on seeding (what was skipped and why)
- allows referencing old seed data in new seed data

## Links to Tickets or other pull requests
Base links to copy
- https://github.com/dBildungsplattform/dbildungs-iam-server/pulls
- https://ticketsystem.dbildungscloud.de/browse/SPSH-594

## Approval for review
- [ ] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

### CODE REVIEW

File: src/console/dbseed/domain/db-seed.service.ts [hunk 1]
```diff
@@ -123,22 +125,31 @@ export class DbSeedService {
...
         for (const file of files) {
+            const serviceProviderUUIDs: string[] = [];
+            for (const spId of file.serviceProviderIds) {
+                const sp: ServiceProvider<true> = await this.getReferencedServiceProvider(spId);
+                serviceProviderUUIDs.push(sp.id);
+            }
             const rolle: Rolle<false> = this.rolleFactory.createNew(
                 file.name,
-                this.getReferencedOrganisation(file.administeredBySchulstrukturknoten).id,
+                (await this.getReferencedOrganisation(file.administeredBySchulstrukturknoten)).id,
```
Comment (marode-cap) [inline src/console/dbseed/domain/db-seed.service.ts:135]: Move this into it's own variable for readability

File: src/console/dbseed/domain/db-seed.service.ts [hunk 2]
```diff
@@ -155,15 +166,23 @@ export class DbSeedService {
                 file.target,
                 file.url,
                 file.kategorie,
-                this.getReferencedOrganisation(file.providedOnSchulstrukturknoten).id,
+                (await this.getReferencedOrganisation(file.providedOnSchulstrukturknoten)).id,
```
Comment (marode-cap) [inline src/console/dbseed/domain/db-seed.service.ts:169]: Move this into it's own variable for readability

File: src/console/dbseed/domain/db-seed.service.ts [hunk 3]
```diff
@@ -237,9 +260,9 @@ export class DbSeedService {
                 undefined,
                 new Date(),
                 new Date(),
-                this.getReferencedPerson(file.personId).id,
-                this.getReferencedOrganisation(file.organisationId).id,
-                this.getReferencedRolle(file.rolleId).id,
+                (await this.getReferencedPerson(file.personId)).id,
+                (await this.getReferencedOrganisation(file.organisationId)).id,
+                (await this.getReferencedRolle(file.rolleId)).id,
```
Comment (marode-cap) [inline src/console/dbseed/domain/db-seed.service.ts:265]: Move these into their own variables for readability

File: src/console/dbseed/db-seed.console-mocked-db-seed-repo.spec.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    let dbSeedRepoMock: DeepMocked<DbSeedRepo>;
+
+    beforeAll(async () => {
+        module = await Test.createTestingModule({
+            imports: [
+                ConfigTestModule,
+                OrganisationModule,
+                KeycloakAdministrationModule,
+                MapperTestModule,
+                DatabaseTestModule.forRoot({ isDatabaseRequired: true }),
```
Comment (marode-cap) [inline src/console/dbseed/db-seed.console-mocked-db-seed-repo.spec.ts:41]: This test uses the database, file should end in `*.integration-spec.ts`

**Review by marode-cap — CHANGES_REQUESTED** [review]

File: src/console/dbseed/repo/db-seed.repo.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
+import { Injectable } from '@nestjs/common';
+import { EntityManager, Loaded, RequiredEntityData } from '@mikro-orm/core';
+import { DbSeed } from '../domain/db-seed.js';
+import { DbSeedEntity } from './db-seed.entity.js';
+
+/**
+ * @deprecated Not for use outside of repo, export will be removed at a later date
+ */
+export function mapAggregateToData(dbSeed: DbSeed<boolean>): RequiredEntityData<DbSeedEntity> {
```
[the remark this answers is not in the document]
Comment (DPDS93CT) [reply src/console/dbseed/repo/db-seed.repo.ts:9]: caught

**Review by marode-cap — APPROVED** [review]

**Pull request merged**